server.js

```diff
@@ -21,9 +16,56 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(endpoints);
+});
+
+app.get("/books", (req, res) => {
+  const showTitle = req.query.title;
+
+  if (showTitle) {
+    const titleSearch = booksData.filter((book) =>
+      book.title.toLowerCase().includes(showTitle.toLowerCase())
+    );
+
+    if (titleSearch.length > 0) {
+      res.json(titleSearch);
+    } else {
+      res.status(404).send("Sorry, there is no book with that title.");
+    }
+  } else {
+    res.json(booksData);
+  }
 });
 
+app.get("/books/:bookId", (req, res) => {
+  const { bookId } = req.params;
+
+  const book = booksData.find((book) => book.bookID === +bookId);
+
+  if (book) {
+    res.json(book);
+  } else {
+    res.status(404).send("Sorry, there is no book with that Id.");
+  }
+});
+
+app.get("/averageratings/:ratingNum", (req, res) => {
+  const { ratingNum } = req.params;
+
+  const rating = booksData.filter(
+    (rating) => Math.round(rating.average_rating) === +ratingNum
```

Comment:
Nice with the average rating. It would be cool if you could filter a range of ratings. like 3-5 or something. Just an idea =)
/Martin